### ISSUE TITLE
Issues using hotkeys with the super key on Wayland

### ISSUE THREAD

**Issue by hollunder** [obsproject/obs-studio#5362]
### Operating System Info

Other

### Other OS

Arch Linux

### OBS Studio Version

27.0.1

### OBS Studio Log URL

https://obsproject.com/logs/ghywsOIX9Ftn7zto

### Expected Behavior

I expect OBS to not react to presses of the Windows key (also called SUPER,  SUPER_L, SUPER_R)  key at all.
I expect OBS to react to the ENTER (RETURN) key the same way as it does react to the SPACE key.

### Current Behavior

OBS by defaults has the keyboard focus on the 'Start Streaming' button.
The button under keyboard focus is triggered whenever either the SPACE or SUPER key is pressed and OBS is in focus.
This also holds true if OBS gets into focus, so for example if you, like me use SUPER+something to switch between workspaces and you switch to a workspace with OBS running, then OBS starts streaming.
This also holds true if the button having focus is not visible, for example if you hide it via View -> Docks -> Controls, in which case you have to un-hide the controls before you can stop the stream again.

The situation is not helped by the fact that OBS under some circumstances keeps keyboard focus when it no longer should. For example when I Click the View menu and while it is open I click on this browser tab here, then the keyboard is still grabbed by OBS and while I think that I am continuing to write this bug report and I press SPACE, as it so often happens while writing a text, OBS starts streaming.

### Steps to Reproduce

1. Start OBS.
2. Press the Windows (SUPER) key or the SPACE key.
3. Watch OBS as it starts streaming.

### Anything else we should know?

Yes. This is a super annoying bug and I have no idea who ever thought it would be a good idea to trigger the keyboard focus action when pressing the Windows key. IMHO they should be keelhauled.

**Comment by mihawk90** [comment]
No button is focused for me when I just open OBS.

Therefore, pressing enter doesn't do anything.

My Super opens the application launcher but obviously that depends on the Desktop Enviroment.

You're also not on the latest version.

**Comment by notr1ch** [comment]
This sounds like an issue with your desktop environment, OBS does nothing special regarding handling of the Windows key.

**Comment by hollunder** [comment]
The ENTER or RETURN key does not work for this, but it would make more sense than the SUPER key.
SUPER and SPACE work.

The keyboard focus is not always visible and even if it is visible it is hard to see. ![keyboard_focus](https://user-images.githubusercontent.com/8207622/135635478-30dd18c0-7ea9-4803-a851-e35d83cc67c4.png)
Even if invisible, the focus is still there.

Arch has not caught up to the latest version released three days ago, but I read the release notes prior to posting this and saw no indication that it has been fixed.
It is an issue that I did not have for years, as far as I can tell it was introduced with OBS 27.

I tried it on two desktop environments with the same result.
Sway on Wayland and i3 on X11.

**Comment by mihawk90** [comment]
> SUPER and SPACE work.

Same result as with the Enter test. Nothing.

There is no focus on any of the buttons.

**Comment by hollunder** [comment]
![hotkeys](https://user-images.githubusercontent.com/8207622/135639605-e22b941d-e1a5-41f9-a147-138dc603287a.png)
I figure it out. I was wrong in thinking that it is related to the keyboard focus.

Above screenshot from Sway/Wayland shows my usual configuration that I entered while running OBS in Sway, SUPER+F11 for stream start, SUPER+F12 for stream stop.
You can't see it? Me neither. But it's there, kind of. Instead of SUPER+F11 what has been stored is just SUPER, as this screenshot taken under i3/X11 demonstrates.
![hotkeys_x11](https://user-images.githubusercontent.com/8207622/135641803-10e135d3-27a4-4e3b-920f-00637af9dc87.png)

When I enter the shortcut under X11 it is stored properly and works under X11 but does nothing under Wayland.

So to summarise, hotkeys entered under Wayland:
1. Don't show up.
2. Their modifier part is still stored and the hotkey is working when OBS has focus.
3. This may lead to unexpected / confusing behavior.

**Comment by RytoEX** [comment]
@kkartaltepe @GeorgesStavracas Is this a Wayland hotkey issue?

**Comment by kkartaltepe** [comment]
It looks like there are a couple issues.

1) Regardless of the compositor, super+ hotkeys dont appear in the hotkey list as mentioned.
2) Assuming the keypress doesnt get eaten by the compositor, obs triggers the hotkey on any super+ key press (on sway super by itself isnt captured so it works, on gnome you need to do super+whatever since it captures super)

**Comment by kkartaltepe** [comment]
Actually

1) The display of hotkeys is just totally broken, since hotkeys only worked when you had obs in focus this is possibly just something not being threaded through.
2) All modifiers appear to behave as individual keys (super/ctrl/etc) and seem to be the only keys recognized. All other keys seem to be ignored, so when you try to press super+x to make a hotkey the dialog only sees super. Trying to make hotkeys without modifiers just doesnt work.

**Comment by Bodyash** [comment]
Same here on Manjaro Gnome 41.1 + Wayland. 

Can't see binded keys in hotkeys configuration.
Can bind only CTRL, SHIFT, SUPER.  

Month ago on Gnome 40 + Wayland and older version of OBS I used NUM_5 key to start/stop recording, now it's impossible.

**Comment by wanwandrew** [comment]
I am having a similar problem in which I cannot bind any keys. I am using Fedora Linux, specifically.

**Comment by kkartaltepe** [comment]
This was implemented in #5650 but do note that you cannot use hotkeys unless obs is in focus.

**Comment by wanwandrew** [comment]
I am aware of this. obs was indeed in focus when I attempted to register
this precise hotkey. Thanks for the help, paki.

**Comment by djnotes** [comment]
I also have  this problem on Fedora 38/Gnome 44.2. Hotkeys do not work in Wayland when the app is not in focus.

**Comment by mihawk90** [comment]
> I also have this problem on Fedora 38/Gnome 44.2. Hotkeys do not work in Wayland when the app is not in focus.

That is an entirely different issue and also working as intended. Wayland has no concept of global hotkeys.

**Comment by djnotes** [comment]
Oh, why is Wayland designed lacking hot keys support?

**Comment by mihawk90** [comment]
That is a question for the Wayland maintainers and way out of scope for OBS issues or support.

**Comment by cfreis** [comment]
I create a a kludge solution for workaround this issue.
You can find it here:
https://github.com/cfreis/obs-scripts
